### PR TITLE
Create endsWith.md

### PR DESCRIPTION
**Pull request by AlstrupJacob**
### Description

created endsWith.md. resolves issue #3724

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/strings/endsWIth/endsWith.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Tags: 
+  - 'String'
+  - 'Comparison'
+  - 'Kotlin'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.endsWith()`** method will compare the end of the sequence with the given suffix, returning a boolean value representing wether they are the same or not.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/endsWIth/endsWith.md:16]:
```suggestion
The **`.endsWith()`** method compares the end of the sequence with the given suffix and returns a boolean value indicating whether they are the same or not.
```

File: content/kotlin/concepts/strings/endsWIth/endsWith.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Syntax
+
+```pseudo
+String.endsWith(suffix)
+```
+
+`String`: The sequence which the suffix is to be compared with.
+
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/endsWIth/endsWith.md:25]:
```suggestion
```
No need for space here

File: content/kotlin/concepts/strings/endsWIth/endsWith.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+`suffix`: a sequence of length `n`, to be compared with the last `n` elements of `String`.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/endsWIth/endsWith.md:26]: After this please explain `ignoreCase` too.

File: content/kotlin/concepts/strings/endsWIth/endsWith.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Example
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/endsWIth/endsWith.md:28]:
```suggestion
## Example

```
Add a space after the heading

File: content/kotlin/concepts/strings/endsWIth/endsWith.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.endsWith()`** method will compare the end of the sequence with the given suffix, returning a boolean value representing wether they are the same or not.
+
+## Syntax
+
+```pseudo
+String.endsWith(suffix)
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/endsWIth/endsWith.md:21]:
```suggestion
String.endsWith(suffix: String, ignoreCase: Boolean = false): Boolean
```
Mention the `ignoreCase` parameter. By default, it stays `false`, which means this function is case-sensitive. However, if we want it to be case-insensitive, we can set `ignoreCase` to `true`.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have requested few changes. Commit then changes and add the requested things so we can move forward to the second review round.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have overall comment too which is about your path of `endsWith.md`. Right now it is `content/kotlin/concepts/strings/endsWIth/endsWith.md` but it should be `content/kotlin/concepts/strings/terms/endsWIth/endsWith.md`. Please fix this too.

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+## Syntax
+
+```pseudo
+String.endsWith(suffix: String, ignoreCase: Boolean = False): Boolean
+```
+
+`String`: The sequence which the suffix is to be compared with.
+`suffix`: a sequence of length `n`, to be compared with the last `n` elements of `String`.
+
+## Example
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:27]:
```suggestion
## Example

```
Add a space here

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+The **`.endsWith()`** method compares the end of the sequence with the given suffix and returns a boolean value indicating wether they are the same or not.
+
+## Syntax
+
+```pseudo
+String.endsWith(suffix: String, ignoreCase: Boolean = False): Boolean
+```
+
+`String`: The sequence which the suffix is to be compared with.
+`suffix`: a sequence of length `n`, to be compared with the last `n` elements of `String`.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:25]: Add info about `ignoreCase` here please after `suffix`.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Have two more changes request from my side. Please commit these changes.

**Comment by dakshdeepHERE** [comment]
@AlstrupJacob have last 2 changes request. Commit them and we will move to 2nd review.

**Comment by AlstrupJacob** [comment]
> @AlstrupJacob have last 2 changes request. Commit them and we will move to 2nd review.

done

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+`ignoreCase`: boolean value dictating whether function is to be case insensitive. Defaults to false.
+
+## Example
+
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:29]:
```suggestion
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  val str = "JohnAppleseed@Hotmail.com"
+  val sfx1 = "@gmail.com"
+  val sfx2 = "@hotmail.com"
+
+  val endsWithSfx1 = str.endsWith { sfx1 }
+  println(endsWithSfx1)
+
+  val endsWithSfx2 = str.endsWith { sfx2, true }
+  println(endsWithSfx2)
+  }
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:43]:
```suggestion
fun main() {
  val str = "JohnAppleseed@Hotmail.com"
  val sfx1 = "@gmail.com"
  val sfx2 = "@hotmail.com"

  val endsWithSfx1 = str.endsWith(sfx1)
  println(endsWithSfx1)

  val endsWithSfx2 = str.endsWith(sfx2, true)
  println(endsWithSfx2)
}
```
Please fix this issue with the code block

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
One last comment from my side. There were some syntax issues with your code example.

**Comment by dakshdeepHERE** [comment]
@AlstrupJacob Looks good for second review.

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+---
+Title: '.endsWith()'
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:2]: The path has an error. The folder name is "endsWIth" but it should be "endsWith"

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
+---
+Title: '.endsWith()'
+Description: 'Returns a boolean value representing wether the sequence ends with the specified suffix.'
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:3]:
```suggestion
Description: 'Returns a boolean value representing whether the sequence ends with the specified suffix.' 

```

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
+---
+Title: '.endsWith()'
+Description: 'Returns a boolean value representing wether the sequence ends with the specified suffix.' 
+Subjects: 
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags: 
+  - 'String'
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:8]:
```suggestion
  - 'Strings'

```

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
+---
+Title: '.endsWith()'
+Description: 'Returns a boolean value representing wether the sequence ends with the specified suffix.' 
+Subjects: 
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags: 
+  - 'String'
+  - 'Comparison'
+  - 'Kotlin'
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:10]:
```suggestion
```

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+Tags: 
+  - 'String'
+  - 'Comparison'
+  - 'Kotlin'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.endsWith()`** method compares the end of the sequence with the given suffix and returns a boolean value indicating wether they are the same or not.
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:16]:
```suggestion
The **`.endsWith()`** method compares the end of the sequence with the given suffix and returns a boolean value indicating whether they are the same or not.

```

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+## Syntax
+
+```pseudo
+String.endsWith(suffix: String, ignoreCase: Boolean = False): Boolean
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:21]:
```suggestion
String.endsWith(suffix, ignoreCase)

```

Comment (caupolicandiaz) [reply content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:21]: generally, data types are detailed with the parameter info below

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+- `String`: The sequence which the suffix is to be compared with.
+- `suffix`: a sequence of length `n`, to be compared with the last `n` elements of `String`.
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:25]:
```suggestion
- `suffix`: A sequence of length `n`, to be compared with the last `n` elements of `String`.

```

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+- `ignoreCase`: boolean value dictating whether function is to be case insensitive. Defaults to false.
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:26]:
```suggestion
- `ignoreCase`: A boolean value dictating whether function is to be case insensitive. Defaults to `false`.

```

File: content/kotlin/concepts/strings/terms/endsWIth/endsWith.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+The example demonstrates the use of `.endsWith()` to compare the end of a sequence, first to the string "@gmail.com", then with the string "@hotmail.com".
+
+```kotlin
+fun main() {
+  val str = "JohnAppleseed@Hotmail.com"
+  val sfx1 = "@gmail.com"
+  val sfx2 = "@hotmail.com"
+
+  val endsWithSfx1 = str.endsWith (sfx1)
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/endsWIth/endsWith.md:38]:
```suggestion
  val endsWithSfx1 = str.endsWith(sfx1)

```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@AlstrupJacob I've added my comments. Please update the entry, and tag me when you've finished the changes. Thanks

**Comment by AlstrupJacob** [comment]
@caupolicandiaz  done

**Review by dakshdeepHERE — APPROVED** [review]

**Review by caupolicandiaz — APPROVED** [review]
@AlstrupJacob Looks good, this is ready to merge

**Pull request merged**